packages/source-iotsitewise/src/alarms/iotevents/util/constructAlarmThreshold.ts

```diff
@@ -0,0 +1,47 @@
+import { COMPARISON_OPERATOR, Threshold } from '@synchro-charts/core';
+import { INTERNAL_ALARM_THRESHOLD_ID } from '@iot-app-kit/core';
+import { ALARM_STATUS, ALARM_STATUS_MAP, AWSUI_RED_600 } from '../constants';
+import { PascalCaseStateName, Alarm } from '../types';
+import { toId } from '../../../time-series-data/util/dataStreamId';
+import { isNumber } from '../../../common/predicates';
+
+export const constructAlarmThresholds = (alarm: Alarm): Threshold[] => {
+  const thresholds: Threshold[] = [];
+
+  const propertyStreamId = toId({ assetId: alarm.assetId, propertyId: alarm.inputPropertyId });
+  const alarmStreamId = toId({ assetId: alarm.assetId, propertyId: alarm.alarmStatePropertyId });
+
+  const alarmStatus = ALARM_STATUS_MAP[alarm.state];
+
+  const inputPropertyThreshold: Threshold = {
+    id: INTERNAL_ALARM_THRESHOLD_ID,
+    comparisonOperator: alarm.comparisonOperator,
+    severity: alarm.severity,
+    value: isNumber(alarm.threshold) ? alarm.threshold : parseFloat(alarm.threshold),
+    dataStreamIds: [ propertyStreamId ],
+    color: AWSUI_RED_600,
+    showValue: true,
+    icon: alarmStatus.icon,
+    description: alarm.rule
+  };
+
+  thresholds.push(inputPropertyThreshold);
+
+  const alarmStatePropertyThresholds = Object.keys(ALARM_STATUS_MAP).map(alarmStatus => {
+    const status = ALARM_STATUS_MAP[alarmStatus as PascalCaseStateName];
+
+    return {
+      value: alarmStatus,
+      color: status.color,
+      severity: status.severity,
+      icon: status.icon,
+      comparisonOperator: COMPARISON_OPERATOR.EQUAL,
+      dataStreamIds: [ alarmStreamId ],
+      description: alarmStatus === ALARM_STATUS.ACTIVE ? alarm.rule : undefined,
+    };
+  });
+
+  thresholds.push(...alarmStatePropertyThresholds);
+
+  return thresholds;
```

Comment:
is there a reason to collect the different things in `thresholds` overtime rather than just returning `[inputPropertyThreshold, ...alarmStatePropertyThresholds]` as the return statement?

Reply:
No I think you are right we can just return the combined thresholds.